src/main/java/seedu/timetable/Timetable.java

```diff
@@ -16,65 +16,65 @@
 
     private ArrayList<Module> modules;
 
-    private Class[] monday = new Class[24];
-    private Class[] tuesday = new Class[24];
-    private Class[] wednesday = new Class[24];
-    private Class[] thursday = new Class[24];
-    private Class[] friday = new Class[24];
-    private Class[] saturday = new Class[24];
-    private Class[] sunday = new Class[24];
+    private TimetableLesson[] monday = new TimetableLesson[24];
+    private TimetableLesson[] tuesday = new TimetableLesson[24];
+    private TimetableLesson[] wednesday = new TimetableLesson[24];
+    private TimetableLesson[] thursday = new TimetableLesson[24];
+    private TimetableLesson[] friday = new TimetableLesson[24];
+    private TimetableLesson[] saturday = new TimetableLesson[24];
+    private TimetableLesson[] sunday = new TimetableLesson[24];
 
-    public TimeTable(int semester) {
+    public Timetable(int semester) {
```

Comment:
since we're using semester, i think should add each sem timetable object to a map (sem -> timetable)
if not at the start we'll have to init 4 sems for 3-4 years worth of timetable objs

Reply:
This semester refers to the sem 1 or sem 2 for the academic year, not the semester for the user (i.e sem 3,4... does not exist)

Reply:
i think sem should be tracked in another class then, cuz the add module function will be based on which sem is currently selected